File: backend/src/main/java/org/backend/payload/response/AnnouncementResponse.java
package org.backend.payload.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AnnouncementResponse {
    private Long id;
    private String title;
    private String content;
    private String thumbnailUrl;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Long officialId;
    private String officialName;
} 

File: frontend_mobile/app/src/main/java/com/example/barangay360_mobile/api/models/AnnouncementResponse.kt
package com.example.barangay360_mobile.api.models

import com.google.gson.annotations.SerializedName
import java.time.OffsetDateTime // Use OffsetDateTime for proper timezone handling

data class AnnouncementResponse(
    @SerializedName("id") val id: Long?,
    @SerializedName("title") val title: String?,
    @SerializedName("content") val content: String?,
    @SerializedName("thumbnailUrl") val thumbnailUrl: String?,
    @SerializedName("createdAt") val createdAt: OffsetDateTime?, // Changed to OffsetDateTime
    @SerializedName("updatedAt") val updatedAt: OffsetDateTime?, // Changed to OffsetDateTime
    @SerializedName("officialId") val officialId: Long?,
    @SerializedName("officialName") val officialName: String?
)